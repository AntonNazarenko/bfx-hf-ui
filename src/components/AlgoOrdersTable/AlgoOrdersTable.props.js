import PropTypes from 'prop-types'

export const propTypes = {
  cancelOrder: PropTypes.func.isRequired,
  filtredAO: PropTypes.array,
  apiClientState: PropTypes.number.isRequired,
}

export const defaultProps = {
  orders: [],
}
