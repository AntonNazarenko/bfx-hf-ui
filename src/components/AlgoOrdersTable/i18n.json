{
    "ru-RU": {
        "name": "Имя",
        "exchange":"Обменник",
        "context": "Контекст",
        "created": "Созданно",
        "symbol": "Символ",
        "label": "Метка"
    },
    "en-EN": {
        "name": "Name",
        "exchange":"Exchange",
        "context": "Context",
        "created": "Created",
        "symbol": "Symbol",
        "label": "Label"
    }
}