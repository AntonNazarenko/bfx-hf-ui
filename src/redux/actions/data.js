import WSActions from './ws'

function syncCandles (symbol, tf, range) {
  const from = +range[0]
  const to = +range[1]

  return WSActions.send(['get.candles', symbol, tf, 'trade', from, to])
}

function getBTs () {
  return WSActions.send(['get.bts'])
}

export default {
  syncCandles,
  getBTs
}
