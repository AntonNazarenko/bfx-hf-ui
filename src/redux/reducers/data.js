import { combineReducers } from 'redux'

import apiKey from './apikey'
import bfx from './bfx-data'
import algoOrders from './algo-orders'

const reducers = combineReducers({
  bfx,
  apiKey,
  algoOrders,
})

export default reducers
