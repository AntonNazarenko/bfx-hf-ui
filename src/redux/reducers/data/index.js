import { combineReducers } from 'redux'

import algoOrders from './algo_orders'

const reducers = combineReducers({
  algoOrders,
})

export default reducers
