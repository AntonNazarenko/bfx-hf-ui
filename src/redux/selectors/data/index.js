import getAlgoOrders from './get_algo_orders'

export {
  getAlgoOrders,
}
