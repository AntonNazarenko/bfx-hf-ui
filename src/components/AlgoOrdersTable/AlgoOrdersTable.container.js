import { connect } from 'react-redux'
import Debug from 'debug'

import { getAPIClientState, getAuthToken } from '../../redux/selectors/ws'
import WSActions from '../../redux/actions/ws'
import GAActions from '../../redux/actions/google_analytics'
import AlgoOrdersTable from './AlgoOrdersTable'

const debug = Debug('hfui:c:algo-orders-table')

const mapStateToProps = (state = {}, ownProps = {}) => {
  const { exID } = ownProps

  return {
    apiClientState: getAPIClientState(state, exID),
    authToken: getAuthToken(state),
    lang: state.ui.lang,
  }
}

const mapDispatchToProps = dispatch => ({
  cancelOrder: (authToken, order) => {
    const { gid, exID } = order

    debug('cancelling algo order %d', +gid)
    dispatch(WSActions.send(['algo_order.cancel', authToken, exID, `${gid}`]))
  },
  gaCancelOrder: () => {
    dispatch(GAActions.cancelAO())
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(AlgoOrdersTable)
