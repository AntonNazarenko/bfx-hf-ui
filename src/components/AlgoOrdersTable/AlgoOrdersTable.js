import React from 'react'
import _capitalize from 'lodash/capitalize'
import { Icon } from 'react-fa'

import { propTypes, defaultProps } from './AlgoOrdersTable.props'
import './style.css'

import i18n from './i18n.json'

export default class AlgoOrdersTable extends React.PureComponent {
  static propTypes = propTypes
  static defaultProps = defaultProps

  cancelOrder(ao) {
    const { cancelOrder, authToken, gaCancelOrder } = this.props
    cancelOrder(authToken, ao)
    gaCancelOrder()
  }

  render() {
    const {
      orders = [], apiClientState, lang,
    } = this.props
    const dictionary = i18n[lang]
    return (
      <ul className='hfui-ao-list__wrapper'>
        {orders.map(ao => (
          <li key={ao.gid} className='hfui-ao-list__entry'>
            {apiClientState === 2 && (
              <ul className='hfui-ao-list__entry-controls'>
                <li>
                  <Icon
                    role='button'
                    tabIndex={0}
                    name='stop'
                    onClick={() => { this.cancelOrder(ao) }}
                    onKeyPress={() => { this.cancelOrder(ao) }}
                  />
                </li>
              </ul>
            )}

            <div className='hfui-ao-list__entry-row-status green' />

            <div className='hfui-ao-list__entry-row'>
              <div className='hfui-ao-list__entry-row-elm'>
                <p className='hfui-ao-list__entry-row-elm-value'>{ao.name}</p>
                <p className='hfui-ao-list__entry-row-elm-label'>{dictionary.name}</p>
              </div>

              <div className='hfui-ao-list__entry-row-elm'>
                <p className='hfui-ao-list__entry-row-elm-value'>{_capitalize(ao.exID)}</p>
                <p className='hfui-ao-list__entry-row-elm-label'>{dictionary.exchange}</p>
              </div>

              <div className='hfui-ao-list__entry-row-elm'>
                <p className='hfui-ao-list__entry-row-elm-value'>{ao.args._margin ? 'Margin' : 'Exchange'}</p>
                <p className='hfui-ao-list__entry-row-elm-label'>{dictionary.context}</p>
              </div>

              <div className='hfui-ao-list__entry-row-elm'>
                <p className='hfui-ao-list__entry-row-elm-value'>{new Date(+ao.gid).toLocaleString()}</p>
                <p className='hfui-ao-list__entry-row-elm-label'>{dictionary.created}</p>
              </div>
            </div>

            <div className='hfui-ao-list__entry-row'>
              <div className='hfui-ao-list__entry-row-elm'>
                <p className='hfui-ao-list__entry-row-elm-value'>{ao.args.symbol}</p>
                <p className='hfui-ao-list__entry-row-elm-label'>{dictionary.symbol}</p>
              </div>
              <div className='hfui-ao-list__entry-row-elm'>
                <p className='hfui-ao-list__entry-row-elm-value'>{ao.label}</p>
                <p className='hfui-ao-list__entry-row-elm-label'>{dictionary.label}</p>
              </div>
            </div>
          </li>
        ))}
      </ul>
    )
  }
}
