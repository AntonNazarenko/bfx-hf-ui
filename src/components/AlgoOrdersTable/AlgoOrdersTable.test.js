import React from 'react'
import { configure, shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import data from './ao.json'
import AlgoOrdersTable from './AlgoOrdersTable'

configure({ adapter: new Adapter() })

describe('AlgoOrdersTable', () => {
    test('it renders the algo orders table', () => {
        const component = mount((
            <AlgoOrdersTable
                exID={'bitfinex'}
                orders={data}
                cancelOrder={()=>{}}
                apiClientState={2}
            />
        ))
        expect(component).toMatchSnapshot()
    })

    test('cancel order',  (done) => {
        const component = mount((
            <AlgoOrdersTable
                exID={'bitfinex'}
                orders={data}
                cancelOrder={(authToken, ao)=>{
                    expect(ao).toBe(data[0])
                    done()
                }}
                apiClientState={2}
            />
        ))

        component.find('.fa.fa-stop').simulate('click')
    })
})
