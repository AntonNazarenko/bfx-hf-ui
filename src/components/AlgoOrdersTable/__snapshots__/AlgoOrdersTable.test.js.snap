// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlgoOrdersTable it renders the algo orders table 1`] = `
<AlgoOrdersTable
  apiClientState={2}
  cancelOrder={[Function]}
  exID="bitfinex"
  orders={
    Array [
      Object {
        "args": Object {
          "_futures": false,
          "_margin": false,
          "amount": 1,
          "cancelDelay": 1,
          "long": Object {
            "args": Array [
              1,
            ],
            "candlePrice": "close",
            "candleTimeFrame": "1m",
            "type": "ema",
          },
          "longEMAPeriod": 1,
          "longEMAPrice": "CLOSE",
          "longEMATF": "1m",
          "longMAPeriod": 0,
          "longMAPrice": "CLOSE",
          "longMATF": "1m",
          "longType": "EMA",
          "orderPrice": 0,
          "orderType": "MARKET",
          "short": Object {
            "args": Array [
              0,
            ],
            "candlePrice": "close",
            "candleTimeFrame": "1m",
            "type": "ema",
          },
          "shortEMAPeriod": 0,
          "shortEMAPrice": "CLOSE",
          "shortEMATF": "1m",
          "shortMAPeriod": 0,
          "shortMAPrice": "CLOSE",
          "shortMATF": "1m",
          "shortType": "EMA",
          "submitDelay": 2,
          "symbol": "tBTCUSD",
        },
        "exID": "bitfinex",
        "gid": "1585330269533",
        "label": "MA Crossover | 1 @ MARKET  | long EMA(1) | short EMA(0)",
        "name": "MA Crossover",
      },
    ]
  }
>
  <ul
    className="hfui-ao-list__wrapper"
  >
    <li
      className="hfui-ao-list__entry"
      key="1585330269533"
    >
      <ul
        className="hfui-ao-list__entry-controls"
      >
        <li>
          <Icon
            Component="span"
            name="stop"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="fa fa-stop"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            />
          </Icon>
        </li>
      </ul>
      <div
        className="hfui-ao-list__entry-row-status green"
      />
      <div
        className="hfui-ao-list__entry-row"
      >
        <div
          className="hfui-ao-list__entry-row-elm"
        >
          <p
            className="hfui-ao-list__entry-row-elm-value"
          >
            MA Crossover
          </p>
          <p
            className="hfui-ao-list__entry-row-elm-label"
          >
            Name
          </p>
        </div>
        <div
          className="hfui-ao-list__entry-row-elm"
        >
          <p
            className="hfui-ao-list__entry-row-elm-value"
          >
            Bitfinex
          </p>
          <p
            className="hfui-ao-list__entry-row-elm-label"
          >
            Exchange
          </p>
        </div>
        <div
          className="hfui-ao-list__entry-row-elm"
        >
          <p
            className="hfui-ao-list__entry-row-elm-value"
          >
            Exchange
          </p>
          <p
            className="hfui-ao-list__entry-row-elm-label"
          >
            Context
          </p>
        </div>
        <div
          className="hfui-ao-list__entry-row-elm"
        >
          <p
            className="hfui-ao-list__entry-row-elm-value"
          >
            3/27/2020, 7:31:09 PM
          </p>
          <p
            className="hfui-ao-list__entry-row-elm-label"
          >
            Created
          </p>
        </div>
      </div>
      <div
        className="hfui-ao-list__entry-row"
      >
        <div
          className="hfui-ao-list__entry-row-elm"
        >
          <p
            className="hfui-ao-list__entry-row-elm-value"
          >
            tBTCUSD
          </p>
          <p
            className="hfui-ao-list__entry-row-elm-label"
          >
            Symbol
          </p>
        </div>
        <div
          className="hfui-ao-list__entry-row-elm"
        >
          <p
            className="hfui-ao-list__entry-row-elm-value"
          >
            MA Crossover | 1 @ MARKET  | long EMA(1) | short EMA(0)
          </p>
          <p
            className="hfui-ao-list__entry-row-elm-label"
          >
            Label
          </p>
        </div>
      </div>
    </li>
  </ul>
</AlgoOrdersTable>
`;
